nexus/tests/integration_tests/instances.rs

```diff
@@ -1611,6 +1611,167 @@ async fn test_instances_invalid_creation_returns_bad_request(
     ));
 }
 
+#[nexus_test]
+async fn test_instance_resize(cptestctx: &ControlPlaneTestContext) {
+    let client = &cptestctx.external_client;
+    let nexus = &cptestctx.server.server_context().nexus;
+    let instance_name = "elastic-band";
+
+    create_project_and_pool(&client).await;
+
+    // Create a stopped instance.
+    let instance = create_instance_with(
+        client,
+        PROJECT_NAME,
+        instance_name,
+        &params::InstanceNetworkInterfaceAttachment::None,
+        vec![],
+        vec![],
+        false,
+    )
+    .await;
+
+    let instance_id = InstanceUuid::from_untyped_uuid(instance.identity.id);
+    let resize_url =
+        format!("/v1/instances/{}/resize", instance_id.to_string());
+
+    // Resizing the instance right away, while it's still stopped, is legal.
+    let _ = NexusRequest::new(
+        RequestBuilder::new(client, Method::PUT, &resize_url)
+            .body(Some(&params::InstanceResize {
+                ncpus: InstanceCpuCount(8),
+                memory: ByteCount::from_gibibytes_u32(8),
+            }))
+            .expect_status(Some(StatusCode::OK)),
+    )
+    .authn_as(AuthnMode::PrivilegedUser)
+    .execute()
+    .await
+    .unwrap()
+    .parsed_body::<Instance>()
+    .unwrap();
+
```

Comment:
This should also check that if we get the instance, then it has the expected shape.